src/base/snark_intf.ml

```diff
@@ -1402,6 +1402,9 @@ module type Run_basic = sig
        , Proof_inputs.t * 'return_value )
        manual_callbacks
 
+  (* Callback, low-level version of [as_prover] and [exists]. *)
+  val as_prover_manual : int -> field array option -> Field.t array
```

Comment:
Style nit: it's better to use `Staged.stage` so that you get an `int -> (field array option -> Field.t array) Staged.t`, which makes it obvious that you shouldn't pass both arguments at the same time.